### ISSUE TITLE
toLowerCase() causing faulty character in Turkish Windows Computer

### ISSUE THREAD

**Issue by zihniogmen** [flyway/flyway#3616]
###### Which version and edition of Flyway are you using?
Flyway 9.5.1

###### If this is not the latest version, can you reproduce the issue with the latest one as well? (Many bugs are fixed in newer releases and upgrading will often resolve the issue)
This is the latest version.

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin)
Gradle plugin

###### Which database are you using? (Type & version)
Postgres 15.1.1

###### Which operating system are you using?
Windows

###### What did you do? (Please include the content causing the issue, any relevant configuration settings, the SQL statement(s) that failed (if any), and the command you ran)
Used spring boot initializer with flyway added as a dependency. With gradle

###### What did you expect to see?
Flyway to validate my pattern in the application.properties file, which is 
`spring.flyway.ignore-migration-patterns=*:missing`

###### What did you see instead?

`Caused by: org.flywaydb.core.api.FlywayException: Invalid migration state 'ignored'. Valid states are: [*, missing, pending, ıgnored, future]`

The problem lies within these lines, which uses "toLowerCase" to lower case strings. 

https://github.com/flyway/flyway/blob/a0d26ea8530494386503207b093f19ff2e08758b/flyway-core/src/main/java/org/flywaydb/core/api/pattern/ValidatePattern.java#L39

But when using a Windows computer with Turkish language, Turkish uppercase "I" letter is converted to "ı" (without the dot above) instead of "i". 

So **IGNORED** converted to **ıgnored** which causes the issue. Maybe using "toLowerCase(locale)" alternative will help ?

**Comment by zihniogmen** [comment]
By the way, as a workaround I've set the locale to Locale.US to overcome this issue.

```
public static void main(String[] args) {
		Locale.setDefault(Locale.US);
		SpringApplication.run(MyApp.class, args);
}
```

But still, I think it should be fixed.

**Comment by tjs145** [comment]
Thanks for letting us know. This should be fixed in 9.15.0
